### PR TITLE
fix(coderd/healthcheck): add daemon-specific warnings to healthcheck output

### PR DESCRIPTION
**Pull request by johnstcn**
Follow-up to https://github.com/coder/coder/pull/11393
Part of https://github.com/coder/coder/issues/10676

- Sorts provisioner daemons by name ascending in output
- Adds daemon-specific warnings to healthcheck output
- Reword some messages

### CODE REVIEW

File: coderd/healthcheck/provisioner.go [hunk 1]
```diff
@@ -119,16 +137,20 @@ func (r *ProvisionerDaemonsReport) Run(ctx context.Context, opts *ProvisionerDae
 			if r.Severity.Value() < health.SeverityError.Value() {
 				r.Severity = health.SeverityError
 			}
-			r.Warnings = append(r.Warnings, health.Messagef(health.CodeUnknown, "Provisioner daemon %q reports invalid API version: %s", daemon.Name, err.Error()))
+			r.Warnings = append(r.Warnings, health.Messagef(health.CodeUnknown, "Some provisioner daemons reports invalid API version information"))
```
Comment (johnstcn) [inline coderd/healthcheck/provisioner.go:140]:
```suggestion
			r.Warnings = append(r.Warnings, health.Messagef(health.CodeUnknown, "Some provisioner daemons report invalid API version information"))
```

File: coderd/healthcheck/provisioner.go [hunk 2]
```diff
@@ -26,7 +27,13 @@ type ProvisionerDaemonsReport struct {
 	Dismissed bool             `json:"dismissed"`
 	Error     *string          `json:"error"`
 
-	ProvisionerDaemons []codersdk.ProvisionerDaemon `json:"provisioner_daemons"`
+	Items []ProvisionerDaemonsReportItem `json:"items"`
```
Comment (mtojek) [inline coderd/healthcheck/provisioner.go:30]: In theory, this is API breaking, right?

File: coderd/healthcheck/provisioner.go [hunk 3]
```diff
@@ -86,6 +93,12 @@ func (r *ProvisionerDaemonsReport) Run(ctx context.Context, opts *ProvisionerDae
 		r.Error = ptr.Ref("error fetching provisioner daemons: " + err.Error())
 		return
 	}
+
+	// Ensure stable order for display and for tests
+	sort.Slice(daemons, func(i, j int) bool {
+		return daemons[i].Name < daemons[j].Name
```
Comment (mtojek) [inline coderd/healthcheck/provisioner.go:99]: doublecheck: we don't need to sort by other fields? There is no way that the report will contain old entries for provisioners with the same name?

File: coderd/healthcheck/provisioner.go [hunk 4]
```diff
@@ -96,19 +109,24 @@ func (r *ProvisionerDaemonsReport) Run(ctx context.Context, opts *ProvisionerDae
...
+			Warnings:          make([]health.Message, 0),
+		}
 
 		// For release versions, just check MAJOR.MINOR and ignore patch.
 		if !semver.IsValid(daemon.Version) {
 			if r.Severity.Value() < health.SeverityError.Value() {
 				r.Severity = health.SeverityError
 			}
-			r.Warnings = append(r.Warnings, health.Messagef(health.CodeUnknown, "Provisioner daemon %q reports invalid version %q", opts.CurrentVersion, daemon.Version))
+			r.Warnings = append(r.Warnings, health.Messagef(health.CodeUnknown, "Some provisioner daemons report invalid version information."))
```
Comment (mtojek) [inline coderd/healthcheck/provisioner.go:122]: nit: maybe we should move all messages to a single file

**Review by mtojek — APPROVED** [review]
Left a few comments to address/respond to. Otherwise, it is 👍

File: coderd/healthcheck/provisioner.go (hunk 1 above)
Comment (mafredri) [reply coderd/healthcheck/provisioner.go:140]: reportED?

File: coderd/healthcheck/provisioner.go (hunk 2 above)
Comment (johnstcn) [reply coderd/healthcheck/provisioner.go:30]: Yeah in theory, but the modifications were only merged this morning :-)

File: coderd/healthcheck/provisioner.go (hunk 3 above)
Comment (johnstcn) [reply coderd/healthcheck/provisioner.go:99]: The database schema requires each provisioner daemon to have a unique name.

**Review by mafredri — APPROVED** [review]

File: coderd/healthcheck/provisioner.go (hunk 1 above)
Comment (johnstcn) [reply coderd/healthcheck/provisioner.go:140]: I think present tense is clearer here.

File: coderd/healthcheck/provisioner.go (hunk 4 above)
Comment (johnstcn) [reply coderd/healthcheck/provisioner.go:122]: https://github.com/coder/coder/issues/11493

**Pull request merged**